### PR TITLE
Support sources as `Uint8Array`s

### PR DESCRIPTION
**Pull request by blitz-1306**
## Changes
- [x] Update dependencies.
  - [x] Support [solc-typed-ast `v18.0.0`](https://github.com/Consensys/solc-typed-ast/releases/tag/v18.0.0) where Solidity sources are represented by `Uint8Array`s.
  - [x] Fix linting issues, caused by linter update.
  - [x] Fix test sample due to compiler version update.
- [x] Autofix to IDE settings file.
- [x] Introduce `SourceFile.rawContents` to store file contents as `Uint8Array`. Tune affected code.

> [!NOTE]
> This PR also resolved #252

Regards.

### CODE REVIEW

File: src/instrumenter/utils.ts [hunk 1]
```diff
@@ -533,3 +533,12 @@ export function getTypeDesc(type: TypeNode): string {
...
 }
+
+export function isASCII(s: string): boolean {
+    for (let i = 0; i < s.length; i++) {
+        if (s.charCodeAt(i) > 127) {
+            return false;
+        }
+    }
+    return true;
+}
```
Comment (blitz-1306) [inline src/instrumenter/utils.ts:544]: Maybe we can use something faster and shorter, for example: https://stackoverflow.com/a/14313213/2269749?

File: src/spec-lang/expr_parser_header.ts [hunk 2]
```diff
@@ -75,24 +76,83 @@ export type ExprParseOptions = {
...
+    tail: Array<[BinaryOperator, SNode, PegsRange]>,
+    headRange: PegsRange,
+    opts: ExprParseOptions): SNode
+{
+    return tail.reduce((acc, [curOp, curVal, curLoc]) =>
+        new SBinaryOperation(acc, curOp, curVal, makeRange({start: headRange.start, end: curLoc.end})), head);
 }
 
-export function parseAnnotation(str: string, ctx: ASTNode, inference: InferType, file: SourceFile, baseOff: number): SAnnotation {
+const seen = new Set();
```
Comment (blitz-1306) [inline src/spec-lang/expr_parser_header.ts:93]: I hope we would not get memory issues with this, as we never reset it. Maybe we should...

**Review by blitz-1306 — COMMENTED** [review]
A few notices

File: src/spec-lang/expr_parser_header.ts (hunk 2 above)
Comment (cd1m0) [reply src/spec-lang/expr_parser_header.ts]: Ooops, that was debugging only. Removed in [0d0da14](https://github.com/Consensys/scribble/pull/253/commits/0d0da1481ae42595676f591d33fb581891ae5846)

File: src/instrumenter/utils.ts (hunk 1 above)
Comment (cd1m0) [reply src/instrumenter/utils.ts]: Fixed in [0d0da14](https://github.com/Consensys/scribble/pull/253/commits/0d0da1481ae42595676f591d33fb581891ae5846)

**Review by cd1m0 — APPROVED** [review]
lgtm

**Pull request merged**